client/backoff/backoff.go

```diff
@@ -64,62 +64,49 @@ func (b *Backoff) Reset() {
 // It will backoff if the function returns a retryable error.
 // Once the context is done, retries will end and the most recent error will be returned.
 // Backoff is not reset by this function.
-func (b *Backoff) Retry(ctx context.Context, f func() error) error {
+func (b *Backoff) Retry(ctx context.Context, f func() error, retryableCodes ...codes.Code) error {
 	// If the context is already done, don't make any attempts to call f.
 	if ctx.Err() != nil {
 		return ctx.Err()
 	}
 
 	// Try calling f while the error is retryable and ctx is not done.
 	for {
-		if err := f(); IsRetryable(err) {
-			select {
-			case <-time.After(b.Duration()):
-				continue
-			case <-ctx.Done():
-				return err
-			}
+		if err := f(); !IsRetryable(err, retryableCodes...) {
```

Comment:
I'd prefer seeing `err != nil && !IsRetryable(err, retryableCodes...)` here, see rationale in another comment below.

Reply:
I find `&&` and `!` statements difficult to reason about. 
`IsRetryable` with the fast path seems easier to read to me.

Reply:
In point of fact, the check should be:
`err == nil || !IsRetryable(err)`